code/controllers/Processes/lighting.dm

```diff
@@ -11,7 +11,7 @@
 
 	for(var/datum/light_source/L in lighting_controller.lights)
 		if(L && L.check())
-			lighting_controller.lights.Remove(L)
```

Comment:
Don't you have to keep the Remove because this doesn't remove the ref otherwise?